src/Sylius/Bundle/CoreBundle/Resources/config/validation/ChannelPricing.xml

```diff
@@ -35,7 +43,7 @@
             </constraint>
         </property>
         <property name="channelCode">
-            <constraint name="NotBlank">
+            <constraint name="NotNull">
```

Comment:
Guessing we're missing a test what happens with a configuration like

> ['channelPricings' => ['' => []]];

just to be sure it won't blow up

Reply:
There is ExistingChannelCodeValidator
```php
if ($value === null) {
    return;
}

if ($this->channelRepository->findOneByCode($value) === null) {
    $this->context->buildViolation($constraint->message)
        ->setParameter('{{ channelCode }}', $value)
        ->addViolation()
    ;
}
```

Reply:
So in case when it's an empty string we're unnecessarily calling the database.

Reply:
Okay, you're right. I'll revert to using NotBlank, and in the ExistingChannelCodeValidator, I'll add a return statement for empty value.